src/js/other-websites/inpage_menu.js

```diff
@@ -51,226 +52,615 @@ async function isUserSignedIn() {
+        height: fxRelayMenuBody.scrollHeight,
+      });
+
+      const searchResults = document.querySelector(
+        ".fx-relay-menu-masks-search-results"
+      );
+      const searchResultsList = searchResults.querySelector("ul");
+
+      // Check if user may have custom domain masks
+      const { premiumSubdomainSet } = await browser.storage.local.get(
+        "premiumSubdomainSet"
+      );
+
+      // API Note: If a user has not registered a subdomain yet, its default stored/queried value is "None";
+      const isPremiumSubdomainSet = premiumSubdomainSet !== "None";
+
+      const masks = await getMasks({
+        fetchCustomMasks: isPremiumSubdomainSet,
+      });
+
+      // Process the masks list:
+      if (masks.length === 0) {
+        // TODO: Add style/class to remove border-radius from header/footer sections
+
+        const search = document.querySelector(".fx-relay-menu-masks-search");
+        search.remove();
+
+        fxRelayMenuBody.classList.remove("is-loading");
+
+      } else if (masks.length > 5) {
+
+        // If there's at least 6 masks, show the search bar
+        await populateMaskList(searchResults, masks, {
+          replaceMaskAddressWithLabel: true,
+        });
+
+        searchResultsList.style.height = `${searchResultsList.offsetHeight}px`;
+
+        const filterSearchForm = document.querySelector(
+          ".fx-relay-menu-masks-search-form"
+        );
+
+        const filterSearchInput = filterSearchForm.querySelector(
+          ".fx-relay-menu-masks-search-input"
+        );
+
+        filterSearchForm.addEventListener("submit", (event) => {
+          event.preventDefault();
+          filterSearchInput.blur();
+        });
+
+        filterSearchInput.addEventListener("input", (event) => {
+          applySearchFilter(event.target.value);
+        });
+
+      } else {
+        // User has between 1-5 masks. Display all of them, but
+        // do not show the search input/filter.
+
+        await populateMaskList(searchResults, masks, {
+          replaceMaskAddressWithLabel: true,
+        });
+
+        const filterSearchForm = document.querySelector(
+          ".fx-relay-menu-masks-search-form"
+        );
+
+        const filterSearchResults = document.querySelector(
+          ".fx-relay-menu-masks-search-results"
+        );
+
+        filterSearchForm.remove();
+        filterSearchResults.classList.add("t-no-search-bar");
+      }
+
+      // Resize iframe
+      await browser.runtime.sendMessage({
+        method: "updateIframeHeight",
+        height: fxRelayMenuBody.scrollHeight,
+      });
+
+      fxRelayMenuBody.classList.remove("is-loading");
+
+      const generateAliasBtn = document.querySelector(
+        ".fx-relay-menu-generate-alias-btn"
+      );
+
+      // Set Generate Mask button
+      await buildContent.components.generateMaskButton();
+
+      // Resize iframe
+      await browser.runtime.sendMessage({
+        method: "updateIframeHeight",
```

Comment:
We're calling this twice in a row? Do we need to call this here AND above?

Reply:
Removed in 72a0b6897317b4c0f420202b273dff4f73da538a